src/deploy/functions/release/fabricator.ts

```diff
@@ -0,0 +1,463 @@
+    } else if (endpoint.platform === "gcfv2") {
+      await this.createV2Function(endpoint);
+    } else {
+      assertExhaustive(endpoint.platform);
+    }
+
+    await this.setTrigger(endpoint);
+  }
+
+  async updateEndpoint(update: planner.EndpointUpdate, scraper: SourceTokenScraper): Promise<void> {
+    if (update.deleteBeforeUpdate) {
+      await this.deleteEndpoint(update.deleteBeforeUpdate);
+      await this.createEndpoint(update.endpoint, scraper);
+      return;
+    }
+
+    if (update.endpoint.platform === "gcfv1") {
+      await this.updateV1Function(update.endpoint, scraper);
+    } else if (update.endpoint.platform === "gcfv2") {
+      await this.updateV2Function(update.endpoint);
+    } else {
+      assertExhaustive(update.endpoint.platform);
+    }
+
+    await this.setTrigger(update.endpoint);
+  }
+
+  async deleteEndpoint(endpoint: backend.Endpoint): Promise<void> {
+    await this.deleteTrigger(endpoint);
+    if (endpoint.platform === "gcfv1") {
+      await this.deleteV1Function(endpoint);
+    } else {
+      await this.deleteV2Function(endpoint);
+    }
+  }
+
+  async createV1Function(endpoint: backend.Endpoint, scraper: SourceTokenScraper): Promise<void> {
+    const apiFunction = gcf.functionFromEndpoint(endpoint, this.sourceUrl);
+    apiFunction.sourceToken = await scraper.tokenPromise();
+    await this.functionExecutor
+      .run(async () => {
+        const op: { name: string } = await gcf.createFunction(apiFunction);
+        await poller.pollOperation<unknown>({
+          ...gcfV1PollerOptions,
+          pollerName: `create-${endpoint.region}-${endpoint.id}`,
+          operationResourceName: op.name,
+          onPoll: scraper.poller,
+        });
+      })
+      .catch(rethrowAs(endpoint, "create"));
+
+    if (backend.isHttpsTriggered(endpoint)) {
+      const invoker = endpoint.httpsTrigger.invoker || ["public"];
+      if (invoker[0] !== "private") {
+        await this.executor
+          .run(async () => {
+            await gcf.setInvokerCreate(endpoint.project, backend.functionName(endpoint), invoker);
+          })
+          .catch(rethrowAs(endpoint, "set invoker"));
+      }
+    }
+  }
+
+  async createV2Function(endpoint: backend.Endpoint): Promise<void> {
+    const apiFunction = gcfV2.functionFromEndpoint(endpoint, this.storage[endpoint.region]);
+
+    // N.B. As of GCFv2 private preview GCF no longer creates Pub/Sub topics
+    // for Pub/Sub event handlers. This may change, at which point this code
+    // could be deleted.
+    const topic = apiFunction.eventTrigger?.pubsubTopic;
+    if (topic) {
+      await this.executor
+        .run(async () => {
+          try {
+            await pubsub.getTopic(topic);
+          } catch (err) {
+            if (err.status !== 404) {
+              throw new FirebaseError("Unexpected error looking for Pub/Sub topic", {
+                original: err as Error,
+              });
+            }
```

Comment:
Missing a return here - this looks like it still tries to create when we get a 404 error?